app/src/main/java/nu/yona/app/api/receiver/YonaReceiver.java

```diff
@@ -152,34 +154,38 @@ private void startService(Context context)
 	}
 
 
-	private void showRestartVPN(final String message)
+	private void showRestartVPN(String message)
 	{
-		try
+		Intent intent = new Intent(context.getApplicationContext(), YonaActivity.class);
+		PendingIntent pendingIntent = PendingIntent.getActivity(context, 0, intent, 0);
+		if (pendingIntent == null)
 		{
-			Intent intent = AppUtils.startVPN(context, true);
-			PendingIntent pIntent = PendingIntent.getActivity(context, (int) System.currentTimeMillis() + 10000, intent, 0);
-
-			Notification notification = new Notification.Builder(context).setContentTitle(context.getString(R.string.appname))
-					.setContentText(message)
-					.setTicker(context.getString(R.string.appname))
-					.setWhen(0)
-					.setVibrate(new long[]{1, 1, 1})
-					.setDefaults(Notification.DEFAULT_SOUND)
-					.setStyle(new Notification.BigTextStyle().bigText(message))
-					.setSmallIcon(R.mipmap.ic_launcher)
-					.setLargeIcon(BitmapFactory.decodeResource(context.getResources(), R.mipmap.ic_launcher))
-					.setContentIntent(pIntent)
-					.setAutoCancel(true)
-					.build();
-
-			notification.flags |= Notification.FLAG_NO_CLEAR;
-
-			NotificationManager notificationManager = (NotificationManager) context.getSystemService(context.NOTIFICATION_SERVICE);
-			notificationManager.notify(0, notification);
+			return;
 		}
-		catch (Exception e)
+		NotificationManager notificationManager = (NotificationManager) context.getSystemService(Context.NOTIFICATION_SERVICE);
+		if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O)
 		{
-			AppUtils.reportException(YonaReceiver.class.getSimpleName(), e, Thread.currentThread());
+			NotificationChannel channel = new NotificationChannel(AppConstant.YONA_VPN_CHANNEL_ID,
```

Comment:
Do we really need to take this in? In the 1.2 version of the app, it doesn't work very well:
* "Click to connect again" opens the application instead of just connecting the VPN again
* Upon opening the app, it connects the VPN again but doesn't remove the "VPN disconnected" notification

So if we don't really have to, I'd rather leave out the VPN notification from the 1.1 release.

Reply:
Agree, with the issue, that It will not remove the notification after VPN connection. 
We can ignore the broadcast for 1.1 and correct the behaviour in 1.2.

Reply:
OK